src/services/NumberService.ts

```diff
@@ -141,6 +141,38 @@ export class NumberService {
             });
     }
 
+    /**
+     * Creates a new plate serial number in the database.
+     * @param attempt - the current number of attempts for generating a Plate Serial Number.
+     * @param awsError - AWS error to be passed when the request fails otherwise null.
+     */
+    public createPlateSerialNumber(attempts: number, awsError: AWSError | null): Promise<PlateSerialNumber> {
+        this.manageAttempts(attempts, awsError);
+        return this.getLastPlateSerialNumber()
+            .then((lastPlateSerialNumber) => {
+                const nextPlateSerialNumberObject = this.createNextPlateSerialNumberObject(lastPlateSerialNumber);
+                const transactExpression = {
+                    ConditionExpression: "plateSerialNumber = :oldPlateSerialNumber",
+                    ExpressionAttributeValues: {
+                        ":oldPlateSerialNumber": lastPlateSerialNumber.plateSerialNumber
+                    }
+                };
+                return this.dbClient.transactWrite(nextPlateSerialNumberObject, transactExpression)
+                    .then(() => {
+                        console.log(`Plate Serial Number Generated successfully`);
+                        return nextPlateSerialNumberObject;
+                    })
+                    .catch((error: AWSError) => {
+                        console.error(error); // limit to 5 attempts
+                        if (error.statusCode === 400) {
+                            console.error(`Attempt number ${attempts} failed. Retrying up to ${Configuration.getInstance().getMaxAttempts()} attempts.`);
+                            return this.createPlateSerialNumber(attempts + 1, error);
```

Comment:
I am thinking refactoring this chunk of code in to something as follows:

 ```
private retry(attempts: number, error: AWSError, numberFunction: any) {
        if (error.statusCode === 400) {
            console.error(`Attempt number ${attempts} failed. Retrying up to ${Configuration.getInstance().getMaxAttempts()} attempts.`);
            return numberFunction(attempts + 1, error);
        }
        throw this.formatAWSError(error);
    }
```

Reply:
was having scope issues when calling the numberFunction...didn't manage to solve it, we can have a look at it later